crates/kitsune_p2p/kitsune_p2p/src/spawn/actor/discover.rs

```diff
@@ -9,43 +8,135 @@ use std::future::Future;
 /// - Err - we were not able to establish a connection within the timeout
 pub(crate) enum PeerDiscoverResult {
     OkShortcut,
-    OkRemote { url: String, con_hnd: MetaNetCon },
+    OkRemote {
+        #[allow(dead_code)]
+        url: String,
+        con_hnd: MetaNetCon,
+    },
     Err(KitsuneP2pError),
 }
 
+pub(crate) trait SearchAndDiscoverPeerConnect: 'static + Send + Sync {
+    fn is_agent_local(
+        &self,
+        agent: Arc<KitsuneAgent>,
+    ) -> MustBoxFuture<'static, KitsuneP2pResult<bool>>;
+
+    fn get_agent_info_signed(
+        &self,
+        agent: Arc<KitsuneAgent>,
+    ) -> MustBoxFuture<'_, Result<Option<AgentInfoSigned>, Box<dyn Send + Sync + std::error::Error>>>;
+}
+
+impl SearchAndDiscoverPeerConnect for Arc<SpaceReadOnlyInner> {
+    fn is_agent_local(
+        &self,
+        agent: Arc<KitsuneAgent>,
+    ) -> MustBoxFuture<'static, KitsuneP2pResult<bool>> {
+        self.i_s.is_agent_local(agent)
+    }
+
+    fn get_agent_info_signed(
+        &self,
+        agent: Arc<KitsuneAgent>,
+    ) -> MustBoxFuture<'_, Result<Option<AgentInfoSigned>, Box<dyn Send + Sync + std::error::Error>>>
+    {
+        self.host_api.get_agent_info_signed(GetAgentInfoSignedEvt {
+            space: self.space.clone(),
+            agent,
+        })
+    }
+}
+
+#[allow(clippy::enum_variant_names)]
```

Comment:
This lint is probably making a fair point, but not a particularly important one I guess

Reply:
I really despise some of clippy's style things. It's more clear to me that the logic is indicating that we "Should" return something, rather than having a vague single word "Return".